src/util.ts

```diff
@@ -195,12 +195,17 @@ export const validateObjectOwner = async (appId: string, address: string, ain: A
   }
 };
 
-export const validateAndGetServiceName = (provider: string): string => {
+export const validateAndGetServiceName = async (
+  provider: string,
+  ainize: Ainize
+): Promise<string> => {
   const serviceName = PROVIDER_SERVICE_NAME_MAP.get(provider);
-  if (!serviceName) {
-    throw new Error('Service is currently not supported');
-  }
-  return serviceName;
+  if (serviceName) return serviceName;
+
+  const service = await ainize.getService(provider);
```

Comment:
currently if provider name not exists in map, it query on ainize.

Reply:
If 'provider' could be service name, how about using 'service nickname' or 'service name alias' instead? i.e., PROVIDER_SERVICE_NAME_MAP -> SERVICE_NICKNAME_MAP or SERVICE_NAME_ALIAS_MAP?

Reply:
I think 'SERVICE_(NICK)NAME_MAP' is good! 
could using "service name" and "provider" be confusing? what about using "service" only?

Reply:
SERVICE_NAME_MAP also looks good to me, thanks!

Reply:
thanks to discuss this issue! 
i will apply renaming in next pr, it's hard to see the changes at a glance. (-> todo)

Reply:
right, sounds great!